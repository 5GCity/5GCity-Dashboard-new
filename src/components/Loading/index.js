/**
 * Loading Component
 * Please write a description
 *
 * @author Guilherme Patriarca <gpatriarca@ubiwhere.com>
 */
import styled from 'styled-components'
import { Loading } from 'element-react'

export default styled(Loading)`
  height: 100vh;

> div:first-child {
  background-color: transparent !important;
}

.el-loading-spinner .path {
  stroke: ${({ theme }) => theme.primaryColor };
  stroke-width: 4;
}
`
