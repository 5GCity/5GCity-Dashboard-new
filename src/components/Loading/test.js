/**
 * Loading Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Loading from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<Loading />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
