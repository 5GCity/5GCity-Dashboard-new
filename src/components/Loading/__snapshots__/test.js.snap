// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: none;
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: rgba(137,151,159,0.15);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 999;
}

.c1 {
  -webkit-animation: rotate 1s linear infinite;
  animation: rotate 1s linear infinite;
  width: 50px;
  height: 50px;
}

.c1 .path {
  stroke-linecap: round;
  -webkit-animation: dash 1.5s ease-in-out infinite;
  animation: dash 1.5s ease-in-out infinite;
}

<div
  className="c0"
>
  <svg
    className="c1"
  >
    <circle
      className="path"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      strokeWidth="4"
    />
  </svg>
</div>
`;
